lib/openrtb2_3/bid.js

```diff
@@ -41,7 +42,30 @@ Bid.prototype.replaceMacros = function(string, valuesMap){
   return replaceMacros(string, valuesMap);
 };
 
-var BidBuilder = function(){};
+Bid.prototype.isBelowFloor = function(bidfloor) {
+  return this.price < bidfloor;
+};
+
+Bid.prototype.hasAdm = function() {
+  // Double ! is intentional
+  return !!this.adm && typeof this.adm === 'string';
+};
+
+Bid.prototype.isBlockedAdomain = function(badv) {
+  if (!Array.isArray(badv)) {
+    throw new Error('badv should be an array');
+  }
+  return badv.some(a => ~this.adomain.indexOf(a));
```

Comment:
Yes by the rtb protocol.
http://www.iab.com/wp-content/uploads/2015/05/OpenRTB_API_Specification_Version_2_3_1.pdf
Page 27